x-pack/plugins/beats_management/server/routes/wrap_route_with_security.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  KibanaRequest,
+  KibanaResponseFactory,
+  RequestHandler,
+  RequestHandlerContext,
+  RouteMethod,
+} from 'src/core/server';
+import { difference } from 'lodash';
+
+export function wrapRouteWithSecurity<P, Q, B>(
+  {
+    requiredLicense = [],
+    requiredRoles = [],
+  }: { requiredLicense?: string[]; requiredRoles?: string[] },
+  handler: RequestHandler<P, Q, B>
+): RequestHandler<P, Q, B> {
+  return async (
+    context: RequestHandlerContext,
+    request: KibanaRequest<P, Q, B, RouteMethod>,
+    response: KibanaResponseFactory
+  ) => {
+    const beatsManagement = context.beatsManagement!;
+    const license = beatsManagement.framework.license;
+    const user = beatsManagement.framework.getUser(request);
+
+    if (
+      requiredLicense.length > 0 &&
+      (license.expired || !requiredLicense.includes(license.type))
+    ) {
+      return response.forbidden({
+        body: {
+          message: `Your ${license.type} license does not support this API or is expired. Please upgrade your license.`,
+        },
+      });
+    }
+
+    if (requiredRoles.length > 0) {
+      if (user.kind !== 'authenticated') {
+        return response.forbidden({
+          body: {
+            message: `Request must be authenticated`,
+          },
+        });
+      }
```

Comment:
I'm not even sure this is necessary since this is the default behavior? Maybe not worth changing the original implementation though.

Reply:
Yea, honestly I tried to reduce the changes as much as possible, so the 'security' handler wrapper may have some redundancy with core's defaults, but I feel like it's not worth doing any additional changes.